### ISSUE TITLE
Certain gear not displaying properly when using gearsets lockstyle function

### ISSUE THREAD

**Issue by zylakitty** [DarkstarProject/darkstar#6062]
**_I have:_**

- [x] searched existing issues (http://github.com/darkstarproject/darkstar/issues/) to see if the issue I am posting has already been addressed or opened by another contributor
- [x] checked the commit log to see if my issue has been resolved since my server was last updated

**_Client Version_** (type `/ver` in game) **:** 
30190430_0

**_Source Branch_** (master/stable) **:** 
no idea what this even means

**_Additional Information_** (Steps to reproduce/Expected behavior) **:** 
I am currently playing on the Nocturnal Souls server, where I have run into this issue.

Certain pieces of gear are not displaying their armor models correctly when specifically using the lockstyle function of the in game gearsets. 

These pieces of gear include the Ceremonial Dress, Novennial Dress, and Novennial Boots. 

When equipping these pieces of gear normally, then using the /lockstyle text command, there seems to be no issue with the models displaying correctly. However when one uses the gearset function to stylelock these pieces there are parts that do not display properly.

Shown in the screenshots below are the armor models being displayed correctly when equipping the items normally, and then comparison screenshots of how they incorrectly appear when using the lockstyle function of the in game gearsets.

The Ceremonial Dress does not display the hand area as it normally should, and will instead overlap whichever piece of gear is actually equipped in that slot, or display the naked model for that slot if nothing is equipped there.

The Novennial Dress has the same error in display as the Ceremonial Dress listed above, and as shown in the screenshots below. The Novennial Boots display as if half of the characters legs are clipping over the armor model.

![Ceremonial Dress Correct display](https://user-images.githubusercontent.com/39413396/58970360-4a655380-8787-11e9-925d-868a67244c8f.png)
![Ceremonial Dress incorrect display](https://user-images.githubusercontent.com/39413396/58970379-53562500-8787-11e9-82b9-94ddbeffa8c4.png)
![Novennial Dress and Boots correct display](https://user-images.githubusercontent.com/39413396/58970392-59e49c80-8787-11e9-9b53-12f3c10766f9.png)
![Novennial Dress and Boots incorrect display](https://user-images.githubusercontent.com/39413396/58970402-60731400-8787-11e9-8ee3-b3731a445625.png)

With all of this said, I do realize this isn't a major issue that warrants priority over game breaking issues, but it is something I would not mind seeing addressed should those working on the game have time, and any efforts are appreciated. Thank you.

**Comment by zylakitty** [comment]
Source Branch (master/stable) : 
Master

Sorry, I got clarification on what this meant.

**Comment by Hozu** [comment]
The issue has to do with gear that locks out other slots. Specifically, that gear needs to set the locked slot to the same (offset?) value as the slot the particular equip is in. Currently, locked slots are being set to whatever is in the hands slot, or the default if nothing is set.

As a workaround for in-game use, actually equip the gear and use /lockstyle on, instead of putting the gear in an equipset and using /lockstyleset.

**Comment by zylakitty** [comment]
Yea, I figured that out in the process of trying to troubleshoot exactly where the issue lay. Thnx for the insight as to what specifically is causing the issue. Should I close this ticket, or leave it to a mod to decide if they want to keep it open?

**Comment by Hozu** [comment]
I'd say leave it open, since there is erroneous behaviour with /lockstyleset.

**Comment by ibm2431** [comment]
Looking at the dress + THF AF hands: Ignoring /lockstyle, does the hands item get properly removed when _equipping_ the dress (either manually or from an equip set)? Or is this issue limited just to the display?

**Comment by zylakitty** [comment]
Yes, this only happens when using the lockstyle feature of the in-game gearsets. If you equip the item normally it will display the armor model as it should, which you can then use the /lockstyle text command to lock the hear appearance.

**Comment by ibm2431** [comment]
**Ignoring /lockstyle**, are equipsets allowing _both_ items _in the set_?

Can one use /equipset to _equip_ both the dress and hands?

**Comment by zylakitty** [comment]
Yea, I figured that out in the process of trying to troubleshoot exactly where the issue lay. Thnx for the insight as to what specifically is causing the issue. Should I close this ticket, or leave it to a mod to decide if they want to keep it open?

**Comment by zylakitty** [comment]
The Dress and the hands are the same item, it's an item that has the descriptor of (cannot equip hand gear), which is in theory where the problem lies?

**Comment by helixhamin** [comment]
Descriptor? No, description don't matter, implemented behavior does. We can remove the reserved slot from the item and allow you to normally equip, as long as the client doesn't freak out.
What he means is, go to an equipset and define both the body and hands, and see if it lets you equip both, even though you should not be able to.

**Comment by teschnei** [comment]
y'all gettin confused

the equipset is using an item that covers two slots.  the server is forgetting to check for multi-slot items when applying an equipset as lockstyle  (ie. only the model id), causing the actual equipment to show through

**Comment by ibm2431** [comment]
@teschnei So the problem is only the _display_, and the server isn't forgetting to check for multi-slot items when _equipping_ an equipset?

**Comment by teschnei** [comment]
if that was the case, the fourth picture wouldn't look like that - those are the default (nothing equipped) mithra hands and legs

**Comment by helixhamin** [comment]
Yes, but we still have the question of the second picture, and why it looks like that.

**Comment by ibm2431** [comment]
I fired up my server to play with it some, some things I've noticed thus far:
A) Equipset is allowing you to _define_ pieces in both slots (this does not on retail)

B) Equipset is not letting you _equip_ both slots, the final piece chosen seems to be either what you have defined in hands or feet slot (what I was wanting to make sure of)

C) When attempting to stylelock an invalid equipset (novennial legs and any defined feet), it will attempt to blend the two items _in the equipset_, not legs + whatever you have currently equipped in feet

D) If you attempt to stylelock with _just_ Novennial legs in the equipset, when currently equipped with any foot item, it will try to blend the legs with your currently equipped foot item

E) If you attempt to stylelock Novennial legs + blank feet in the equip set, when wearing nothing in either slot, it will try to blend Novennial legs with default naked feet

F) E will also occur if you have any non-Novennial legs currently equipped in the legs slot

G) If you attempt to stylelock n=Novennial legs + blank feet, while having Novennial legs _actually equipped_, you end up with the normal looking Novennial legs only, no blended default feet

H) G also occurs if you have a different foot-locking piece _actually equipped_ (Dinner Hose) when you try to stylelock the Novennial Legs, it doesn't have to be the same piece being locked

I) But if your equipset is Novennial Legs + _Remove_ feet (Red X), you will end up with blended Novennial + Default, even if you're currently equipped with a valid Novennial/Dinner legs
